### ISSUE TITLE
.events() isn't always available on DOM

### ISSUE THREAD

**Issue by aronallen** [cyclejs/cycle-dom#84]
Hi
I found out that DOM.events isn't available on a root DOM driver source, this means that a component can work if it is nested in other components, but fails if it is not. So I can run my component as through main, but if I try to run SpinningCounter directly I get an error.
```js
function intent ({DOM, HTTP, ...sources}) {
  const clicks$ = DOM.events('click').tap(e => e.preventDefault()).map(1);
  const seed$ = HTTP.switch()
    .map(r => r.text)
    .map(r => JSON.parse(r))
    .map(e => e.number)
    .startWith(0)
    .publish();
  seed$.connect();
  return _.merge({}, sources, {ticks$ : clicks$, seed$});
}

function model ({ticks$, seed$, ...sources}) {

  const external$ = sources.count$ ? sources.count$ : Observable.just(0);

  const count$ = seed$
  .startWith(0)
  .merge(ticks$)
  .scan((acc, x) => acc + x)
  .combineLatest(external$, (x, y) => x + y);


  return _.merge({}, sources, {count$, HTTP : Observable.just({
    url : `/start.json`,
    query : {
      x : cuid()
    }
  })});
}

function view ({count$, ...sources}) {
  const id = cuid();
  const vtree$ = count$.map(x => h('div', null, h('h1', {
    key : id,
    style : {
      transform : `rotateZ(${x/60 * 360}deg)`,
      transition : 'transform 1s  cubic-bezier(0.035, 0.690, 0.015, 1.160)',
      float : `left`,
      clear : `both`,
      webkitUserSelect : 'none'
    }
  }, `${x}`)));

  return _.merge({}, sources, {
    count$,
    DOM : vtree$
  });
}

export default function SpinningCounter (sources) {
  return view(model(intent(sources)));
}
```
```js
function main ({DOM, HTTP}) {
  const counter1 = isolate(SpinningCounter)({DOM, HTTP});
  const counter2 = isolate(SpinningCounter)({DOM, HTTP});
  const count$ = counter1.count$.combineLatest(counter2.count$, (x, y) => x + y);
  const counter3 = isolate(SpinningCounter)({DOM, HTTP, count$});
  return {
    HTTP : Observable.merge(counter1.HTTP, counter2.HTTP, counter3.HTTP),
    DOM : Observable.just(h('div', null,
      [
        counter1.DOM,
        counter2.DOM,
        counter3.DOM
      ]
  ))
  };
}



//this will work
Cycle.run(main, {
  DOM : makeDOMDriver('body'),
  HTTP : makeHTTPDriver()
});

```

**Comment by aronallen** [comment]
```
//this won't work
Cycle.run(SpinningCounter, {
  DOM : makeDOMDriver('body'),
  HTTP : makeHTTPDriver()
});
```

**Comment by staltz** [comment]
Interesting, I've never thought of using it like that. I guess it's a nice feature to add. :+1:

In the meanwhile, do this instead:
```diff
- DOM.events('click')
+ DOM.select('*').events('click')
```

I actually don't know if `'*'` works, but anyway, the idea is that we should use select instead of grabbing all the clicks.

**Comment by aronallen** [comment]
I found the issue because I was playing around with cycle nested, and then I remembered that any dataflow component is also a full cycle app. 

So I tried to start the component by itself, which didn't work. I could only get it to work by wrapping my h1 in a div. By wrapping it in a div the same component worked both as a dataflow component, and as an individual app, but I don't like the idea of making extra markup to get things to work.
